### PR TITLE
[tinyexr] initial integration

### PR DESCRIPTION
**Pull request by zchcai**
Still waiting for primary contact reply.
Here is the created outreach issue [link](https://github.com/syoyo/tinyexr/issues/134).

### CODE REVIEW

File: projects/tinyexr/build.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# Env settings
+CXX=clang++
```
Comment (inferno-chromium) [inline projects/tinyexr/build.sh:19]: Remove these two line.

File: projects/tinyexr/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER TODO@gmail.com
```
Comment (inferno-chromium) [inline projects/tinyexr/Dockerfile:18]: Remove this line.

File: projects/tinyexr/build.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+# prepare files
+cp test/fuzzer/fuzz.cc $SRC/tinyexr_fuzzer.cc
+
+# build fuzzers
+$CXX $CXXFLAGS -std=c++11 -I. \
+     $SRC/tinyexr_fuzzer.cc -o $OUT/tinyexr_fuzzer \
```
[the remark this answers is not in the document]
Comment (zchcai) [reply projects/tinyexr/build.sh:23]: This fuzzer is located here: https://github.com/syoyo/tinyexr/blob/master/test/fuzzer/fuzz.cc
Are you suggesting we copy it into this CL and make our usage independent of that location?

File: projects/tinyexr/project.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+homepage: "https://github.com/syoyo/tinyexr"
+language: c++
+primary_contact: "TODO@gmail.com"
+fuzzing_engines:
+    - libfuzzer
+sanitizers:
+    - address
```
[the remark this answers is not in the document]
Comment (zchcai) [reply projects/tinyexr/project.yaml:7]: Currently no for this. Also, when tinyexr writes this fuzzer, they didn't use memory:
https://github.com/syoyo/tinyexr/blob/ef9f46664319d6f5fe73a538829744fba3dcb306/test/fuzzer/Makefile#L1

File: projects/tinyexr/build.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (inferno-chromium) [reply projects/tinyexr/build.sh:23]: ah fuzzer is already there, then no need.

File: projects/tinyexr/project.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (inferno-chromium) [reply projects/tinyexr/project.yaml:7]: We shouldnt be using their makefile flags when building, they should use our CFLAGS and CXXFLAGS set in environment which have the right sanitizer options. Their current one mixes both ASan and UBSan in same build, whereas we prefer them seperate. A low severity UBSan bug shouldn't block a high severity ASan bug find.

[the remark this answers is not in the document]
Comment (zchcai) [reply projects/tinyexr/project.yaml:7]: I see. Thanks for the explanation. For this current commit, the "memory" option cannot pass "check_build". Thus, I didn't add this to it now. Maybe later we can try to support this option.

**Comment by jonathanmetzman** [comment]
Closing since it appears tinyexr does not want this integration: https://github.com/syoyo/tinyexr/issues/134

**Pull request closed without merging**